dockerfiles/theia-dev/Dockerfile

```diff
@@ -49,6 +49,8 @@ EXPOSE 3000 3030
 # Configure npm and yarn to use home folder for global dependencies
 RUN npm config set prefix "${HOME}/.npm-global" && \
     echo "--global-folder \"${HOME}/.yarn-global\"" > ${HOME}/.yarnrc && \
+    # Setup node-gyp. Yarn will be failed unless installing required files here.
```

Comment:
AFAIK there is no need to install node-gyp there

Reply:
Yes, here is a considerable point.
As you say, there is no `yarn` call in `theia-dev/Dockerfile`.
But e2e uses `theia-dev` as the base image and calls `yarn`.

And end users that use this image will call `yarn` in the container.
They may encounter build failures.

Just IMO but it's better to preset header files provided node-gyp.

Reply:
I moved `node-gyp install` into `entrypoint.sh` because the home directory will be moved on non-root environments.
The new patch has already pushed.

`node-gyp install` deploys files into `${HOME}/.node-gyp`.
And, as you know, `${HOME}` is ephemeral directory.
So it should be called in every bootstrap.